src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,73 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    bool rc = false;
+    auto transpose = as_type_ptr<op::Transpose>(n);
+
+    if (transpose)
+    {
+        auto perm = as_type_ptr<op::Constant>(n->get_argument(1));
+        auto perm_value = perm->get_vector<int64_t>();
+        auto data = n->get_argument(0);
+        auto data_shape = n->input_value(0).get_partial_shape().to_shape();
+
+        // Data shape should be >= 3
+        if (data_shape.size() < 3 || data_shape.size() > 4)
+            return false;
+
+        // Check if input shape contains a Dim of value 1
+        int index_of_one = 0; // index of dim 1 in data_shape
+        for (int i = 0; i < data_shape.size(); i++)
+        {
+            if (data_shape[i] == 1)
+            {
+                index_of_one = i;
+                break;
+            }
+        }
+
+        if (!index_of_one)
+            return false;
+
+        // Remove index of 1 from perm_value and check if the other
+        // dimensions haven't interchanged
+        vector<int64_t> perm_value_copy = perm_value;
+
+        for (int i = 0; i < perm_value_copy.size(); i++)
+        {
+            if (perm_value_copy[i] == index_of_one)
+            {
+                perm_value_copy.erase(perm_value_copy.begin() + i);
+                break;
+            }
+        }
+
+        // If all the indices are in increasing order,
+        // then the other dims are unchanged and only dim of 1
+        // has changed its position
+        for (int i = 0; i < perm_value_copy.size() - 1; i++)
+        {
+            if (perm_value_copy[i] > perm_value_copy[i + 1])
+                return false; // not a valid permutation
+        }
+
+        Shape out_shape{};
+        AxisVector input_order{};
+        for (int i = 0; i < perm_value.size(); i++)
+        {
+            out_shape.push_back(data_shape[perm_value[i]]);
+            input_order.push_back(perm_value[i]);
+        }
+
+        auto reshape_op = make_shared<op::Reshape>(data, input_order, out_shape);
```

Comment:
The output shape for the Reshape is just the output shape from Transpose. So can we just use `n->output_value(0).get_partial_shape()` ?

Reply:
Was doing the same thing indirectly ...changed it though